### DISCUSSION TITLE
"(N/2) + 1" rule question

### DISCUSSION THREAD

**Discussion by 1Xela1** [rqlite/rqlite#1790] [Q&A] [answered]
Hi.

As I know from documentation:
`For a cluster of N nodes in size to remain operational, at least (N/2)+1 nodes must be up and running, and be in contact with each other.`

But what if I run all nodes with "-raft-cluster-remove-shutdown=true"? 

E.g.
There are 5 working nodes with "-raft-cluster-remove-shutdown=true". If I have correct understanding, if I turn off 3 nodes, then 3 nodes will be removed from cluster and from that moment it will be 2-nodes cluster (and if I turn on it back, I will have 5-nodes cluster again, and all new changes will be replicated to those 3 nodes). It means, system will be in working state, if at least 1 node is working (and "(N/2)+1" rule is not working here, because "N" changes).  Am I right?
Note: Buy saying " turn on node back" I mean run node like this: 
`rqlited.exe -node-id %nodeId% -http-addr %nodeAddress%:4001 -raft-addr %nodeAddress%:4002 -join %mainNode%:4002 -raft-cluster-remove-shutdown=true ~/node` (like it was initially started)

If I am right, can I use Rqilte in that way?
I mean, I want to have 5 nodes. And sometimes I want to turn off from 1 to 4 nodes e.g for a day. Can I be sure, that if I turn on those nodes back, I will get the latest changes(table changes, data changes, etc) from the working node? 

And one more question.
`If the node is shut down gracefully (it receives SIGTERM for example) it will first contact the Leader and remove itself from the cluster, and then the rqlite process will terminate.`
What happens if node (with -raft-cluster-remove-shutdown=true) will be shut down due to power outage?

Thank you.

**Comment by otoolep** [comment, marked as answer]
Your understanding is correct. If that flag is set the node will remove itself from the cluster on shutdown, and thereby reduce the size of the cluster. The Leader shouldn't complain that the node is "gone". Try it yourself and see -- try shutting down a node with and without the flag, and you should see a difference in logs from the Leader (it might take a couple of seconds for the Leader to stop trying to contact the downed node).

Remove-on-shutdown is a best-effort thing however. If the Leader is not contactable, then the node can't remove itself, but will proceed to shut down anyway.

>What happens if node (with -raft-cluster-remove-shutdown=true) will be shut down due to power outage?

Remove-on-shutdown only happens with a graceful shutdown. A power outage implies a sudden exit of the `rqlited` process. Perhaps you want to reap nodes instead.

https://rqlite.io/docs/clustering/general-guidelines/#automatically-removing-failed-nodes
